### PR TITLE
Fix not being able to upload files using string file paths

### PR DESCRIPTION
**Pull request by chrisvanrun**
Ran into this when testing out a helper function.

### CODE REVIEW

File: gcapi/client.py [hunk 1]
```diff
@@ -578,7 +578,7 @@ def _upload_files(self, *, files, **kwargs):
                 uploads.append(
                     (
                         yield from self.__org_api_meta.uploads.upload_fileobj(
-                            fileobj=f, filename=file.name
+                            fileobj=f, filename=f.name
```
Comment (chrisvanrun) [inline gcapi/client.py:581]: This is the actual fix, the rest is tests to cover both `str` and `Path` file uploads.

**Comment by jmsmkn** [comment]
This looks good but extending those tests is really slow, could you instead write a dedicated test for this?

**Comment by chrisvanrun** [comment]
Changes:
* Added some lines about testing in README
* Added a simple bash-level retry for pulling images (seems like the login was the main culprit here)
* Simplified testing for this one change

**Comment by jmsmkn** [comment]
Pinging @pkcakeout to review, especially the retry stuff

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -58,12 +58,27 @@ jobs:
...
-          aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws/diag-nijmegen/grand-challenge/http
-          docker pull public.ecr.aws/diag-nijmegen/grand-challenge/http:latest
-          docker logout public.ecr.aws/diag-nijmegen/grand-challenge/http
+          retry() {
+            local -r -i max_attempts="$1"; shift
+            local -i attempt_num=1
+            until eval "$@"
+            do
+              command_return=$?
+              if ((attempt_num==max_attempts))
```
Comment (pkcakeout) [inline .github/workflows/ci.yml:67]: Would do >= instead of == because that better captures intent.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -58,12 +58,27 @@ jobs:
...
+              then
+                return $command_return
+              else
+                echo "Retrying..."
+                sleep $((attempt_num++ * 10))
+              fi
+            done
+          }
```
Comment (pkcakeout) [inline .github/workflows/ci.yml:75]: I think this is probably correct, but I wonder why we are not using one of the solutions that are available right there - either on the commandline or on the marketplace for GitHub Actions? I found a few from a quick googling in the marketplace - the first one would probably already work:

- https://github.com/marketplace/actions/shell-command-retry-action
- https://github.com/marketplace/actions/retry-action#retry-command
- https://github.com/marketplace/actions/retry-step

My favorite solution however, would be to just use `retry` om thje console:

~~~~~~~~~~~~
root@2e7d0e40a0c9:/# apt install retry
[...]
root@2e7d0e40a0c9:/# retry --help


NAME
  retry - Repeat command until a criteria is met, usually success.

SYNOPSIS
  retry [-v] [-h] [-u until] [-w while] command ...

DESCRIPTION

  The tool repeats the given command until the command is successful,
  backing off with a configurable delay between each attempt.

[...]
~~~~~~~~~~~~

**Comment by pkcakeout** [comment]
I can fix this next week...

File: .github/workflows/ci.yml (hunk 3 above)
Comment (pkcakeout) [reply .github/workflows/ci.yml:75]: PR #130 contains my fix for this...

**Review by jmsmkn — APPROVED** [review]

**Pull request merged**